src/traces/heatmap/calc.js

```diff
@@ -20,6 +18,10 @@ var colorscaleCalc = require('../../components/colorscale/calc');
 var hasColumns = require('./has_columns');
 var convertColumnXYZ = require('./convert_column_xyz');
 var maxRowLength = require('./max_row_length');
+var cleanData = require('./clean_data');
```

Comment:
Oh I see, you generated `cleanZ` to no `'z'` data arrays. 

So maybe `clean_2d_array.js` instead?